modules/data.remote/inst/gee2pecan_s2.py

```diff
@@ -690,27 +718,83 @@ def xr_dataset_to_timeseries(xr_dataset, variables):
+        # (pixels outside the polygon),
         # from snap algorihtm nans occure due to input/output ouf of bounds
         # checking.
         # TODO: flaggging with snap biophys algorith or some other solution to
         # check which nan are from snap    algorithm and which from 1d to 2d transformation
-        nans = np.isnan(xr_dataset[var]).sum(dim=['x', 'y'])
+        nans = np.isnan(xr_dataset[var]).sum(dim=["x", "y"])
         sample_n = len(xr_dataset[var].x) * len(xr_dataset[var].y) - nans
 
         # compute how many of the nans are inside aoi (due to snap algorithm)
-        out_of_aoi_pixels = (len(xr_dataset[var].x) * len(xr_dataset[var].y)
-                             - xr_dataset.aoi_pixels)
+        out_of_aoi_pixels = (
+            len(xr_dataset[var].x) * len(xr_dataset[var].y) - xr_dataset.aoi_pixels
+        )
         nans_inside_aoi = nans - out_of_aoi_pixels
-        df['aoi_nan_percentage'] = nans_inside_aoi / xr_dataset.aoi_pixels
+        df["aoi_nan_percentage"] = nans_inside_aoi / xr_dataset.aoi_pixels
 
-        df[var+'_se'] = df[var+'_std'] / np.sqrt(sample_n)
+        df[var + "_se"] = df[var + "_std"] / np.sqrt(sample_n)
 
     return df
+
+
+def gee2pecan_s2(geofile, outdir, start, end, qi_threshold):
+    """ 
+    Downloads Sentinel 2 data from gee and saves it in a netCDF file at the specified location.
+    
+    Parameters
+    ----------
+    geofile (str) -- path to the file containing the name and coordinates of ROI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date of the data request in the form YYYY-MM-DD
+    
+    qi_threshold (float) -- From satellitetools: Threshold value to filter images based on used qi filter. qi filter holds labels of classes whose percentages within the AOI is summed. If the sum is larger then the qi_threshold, data will not be retrieved for that date/image. The default is 1, meaning all data is retrieved
+        
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory.
+            
+    Python dependencies required: earthengine-api, geopandas, pandas, netCDF4, xarray      
+    """
+
+    # read in the input file containing coordinates
+    df = gpd.read_file(geofile)
+
+    request = S2RequestParams(start, end)
+
+    # filter area of interest from the coordinates in the input file
+    area = AOI(df[df.columns[0]].iloc[0], df[df.columns[1]].iloc[0])
+
+    # calcuate qi attribute for the AOI
+    ee_get_s2_quality_info(area, request)
+
+    # get the final data
+    ee_get_s2_data(area, request, qi_threshold=qi_threshold)
+
+    # convert dataframe to an xarray dataset, used later for converting to netCDF
+    s2_data_to_xarray(area, request)
+
+    area.data = compute_ndvi(area.data)
```

Comment:
I know you already know this, just to document it: we would ideally like to compute ndvi on the GEE side and return with the other bands for S2 too, like we do for the L8.

Reply:
@istfer `gee2pecan_s2()` now calculates NDVI on GEE and all the bands are being retrieved without causing any issue in SNAP. Shall I add the initial book documentation in this PR itself?

Reply:
Hi Ayush, I'll check this PR again as soon as I can but it might have to wait until tomorrow. In the meantime, yes, it would be wonderful if you could already add the initial book documentation to this PR :+1: